provider/pkg/gen/schema.go

```diff
@@ -1386,6 +1386,21 @@ func (g *packageGenerator) genTypeSpec(typeName, propName string, prop *discover
 			Type: "object",
 			Ref:  referencedTypeName,
 		}, nil
+	case prop.Type == "object" && prop.AdditionalProperties != nil:
+		// The prop is a map with a string key and a simple value
+		// if it doesn't have a ref.
+		if prop.AdditionalProperties.Ref == "" {
+			return &schema.TypeSpec{
+				Type: "object",
+				AdditionalProperties: &schema.TypeSpec{
+					Type: "string",
```

Comment:
This should not be hard-coded to `string`. Ideally, this uses `prop.AdditionalProperties.Type` but that leads to some errors that I didn't look deeply into. But then again, perhaps a `string` is ok here for now?

Reply:
it's true that these are usually strings, but I think if you instead recurse on prop.AditionalProperties, you can get the full type.